### PR TITLE
Sm/bump-core

### PR DESCRIPTION
**Pull request by mshanemc**
### What does this PR do?
bumps dependencies (ts to latest that will compile, lint, etc)
handle all linter issues

part of the ripples from @W-9660828@

### CODE REVIEW

File: src/sfdxCommand.ts [hunk 1]
```diff
@@ -145,7 +145,7 @@ export abstract class SfdxCommand extends Command {
   protected flags!: OutputFlags<any>; // eslint-disable-line @typescript-eslint/no-explicit-any
 
   // The parsed args for easy reference by this command; assigned in init
-  protected args!: OutputArgs<any>; // eslint-disable-line @typescript-eslint/no-explicit-any
+  protected args!: OutputArgs;
```
Comment (mshanemc) [inline src/sfdxCommand.ts:148]: this was not used by CLI (and from looking in oclif I don't see that it worked previously).  In any case, I can be persuaded that it's a breaking change.

File: src/sfdxCommand.ts [hunk 2]
```diff
@@ -394,7 +394,7 @@ export abstract class SfdxCommand extends Command {
     }
     // Emit an event for the analytics plugin.  The ts-ignore is necessary
     // because TS is strict about the events that can be emitted on process.
-    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```
Comment (mshanemc) [inline src/sfdxCommand.ts:397]: eslint changed the name of the rule !!

File: src/sfdxFlags.ts [hunk 3]
```diff
@@ -58,9 +58,9 @@ function merge<T>(
 function merge<T>(kind: flags.Kind, flag: IFlag<T>, describable: flags.Describable): flags.Discriminated<flags.Any<T>> {
   if (has(flag, 'validate') && hasFunction(flag, 'parse')) {
     const parse = flag.parse.bind(flag);
-    flag.parse = <T>(val: string, ctx: unknown): T => {
+    flag.parse = <T2>(val: string, ctx: unknown): T2 => {
```
Comment (mshanemc) [inline src/sfdxFlags.ts:61]: no-shadow caught this (levels of `T`)

File: test/unit/sfdxCommand.test.ts [hunk 4]
```diff
@@ -108,10 +112,10 @@ let jsonToStdout: boolean;
 async function mockStdout(test: (outLines: string[]) => Promise<void>) {
   const oldStdoutWriter = process.stdout.write.bind(process.stdout);
   const lines: string[] = [];
-  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
-  // @ts-ignore
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore TODO: why isn't `create` invokable?!
```
Comment (shetzel) [inline test/unit/sfdxCommand.test.ts:116]: Is this TODO comment accurate?

**Review by shetzel — APPROVED** [review]

File: src/sfdxCommand.ts (hunk 1 above)
Comment (WillieRuemmele) [reply src/sfdxCommand.ts:148]: I'd vote against this change. I think we need to be as delicate as possible with this change. Anyone else using `SfdxCommand` could be affected by it... we saw what happened when messing with types in core

File: src/sfdxFlags.ts [hunk 5]
```diff
@@ -648,7 +648,7 @@ function validateCustomFlag<T>(key: string, flag: flags.Any<T>): flags.Any<T> {
   return flag;
 }
 
-function isBuiltin(flag: object): flag is flags.Builtin {
+function isBuiltin(flag: Record<string, unknown>): flag is flags.Builtin {
```
Comment (WillieRuemmele) [inline src/sfdxFlags.ts:651]: this makes me super nervous

File: src/sfdxCommand.ts (hunk 1 above)
Comment (mshanemc) [reply src/sfdxCommand.ts:148]: the underlying oclif/command doesn't have `OutputArgs` as a generic.  That's what prompted the change.  Sorry for omitting that important explanation

File: test/unit/sfdxCommand.test.ts (hunk 4 above)
Comment (mshanemc) [reply test/unit/sfdxCommand.test.ts:116]: no, that was copy pasted from the other one.

File: src/sfdxFlags.ts (hunk 5 above)
Comment (mshanemc) [reply src/sfdxFlags.ts:651]: will revert and ts-comment it.  It's not public, but I can see something untyped getting passed down into it.

File: src/sfdxCommand.ts (hunk 1 above)
Comment (mshanemc) [reply src/sfdxCommand.ts:148]: @mdonnalley changed it.

**Pull request merged**